tools/mtouch/Application.cs

```diff
@@ -1335,6 +1336,49 @@ static int GetIndentation (string line)
 			};
 		}
 
+		// This method generates the manifes that is required by the symbolication in order to be able to debug the application, 
+		// The following is an example of the manifest to be generated:
+		// <mono-debug version=”1”>
+		//	<app-id>com.foo.bar</app-id>
+		//	<build-date>datetime</build-date>
+		//	<build-id>build-id</build-id>
+		//	<build-id>build-id</build-id>
+		// </mono-debug>
+		// where:
+		// 
+		// app-id: iOS/Android/Mac app/package ID. Currently for verification and user info only but in future may be used to find symbols automatically.
+		// build-date: Local time in DateTime “O” format. For user info only.
+		// build-id: The build UUID. Needed for HockeyApp to find the mSYM folder matching the app build. There may be more than one, as in the case of iOS multi-arch.
+		void GenerateMSymManifest (Target target, string target_directory)
+		{
+			var manifestPath = Path.Combine (target_directory, "manifest.xml");
+			if (String.IsNullOrEmpty (target_directory))
+				throw new ArgumentNullException (nameof (target_directory));
+			var root = new XElement ("mono-debug",
+				new XAttribute("version", 1),
+				new XElement ("app-id", BundleId),
+				new XElement ("build-date", DateTime.Now.ToString ("O")));
+
+			foreach (var asm in target.Assemblies) {
+				foreach (var abi in abis) {
+					var macho_file = $"{asm.FileName}.{abi.AsArchString ()}.o";
+					var src = Path.Combine (target.ArchDirectory, macho_file);
+					if (File.Exists (src)) {
+						var data = MachO.Read (macho_file) as MachOFile;
+						// add a node for each uuid cmd 
+						foreach (var cmd in data.load_commands) {
+							if (cmd is UuidCommand) {
+								var uuidCmd = cmd as UuidCommand;
+								root.Add (new XElement ("build-id", Encoding.UTF8.GetString (uuidCmd.uuid)));
+							}
+						}
+					}
+				}
+			}
+			// WriteAllText will replace the file if needed.
+			File.WriteAllText (manifestPath, root.ToString ());
```

Comment:
Use `Driver.WriteIfDifferent` instead to write the contents, which doesn't write anything if the contents haven't changed.